tensorflow/cc/framework/gradients.cc

```diff
@@ -165,6 +194,9 @@ Status SymbolicGradientBuilder::Initialize() {
   // handle computing gradients over small subgraphs from a very large graph.
   pending_.resize(scope_.graph()->num_node_ids(), 0);
   {
+    std::vector<bool> reachable_nodes;
+    GetReachableNodes(&reachable_nodes);
+    
```

Comment:
Can you add a TODO to check that inputs_ are reachable from outputs_ here? (It would be easy to add the check now, but let's save it for another PR since it will require another test case.)